components/brave_rewards/core/endpoints/get_wallet/get_wallet.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/core/endpoints/get_wallet/get_wallet.h"
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "brave/components/brave_rewards/core/endpoint/promotion/promotions_util.h"
+#include "brave/components/brave_rewards/core/rewards_engine_impl.h"
+#include "brave/components/brave_rewards/core/wallet/wallet.h"
+#include "net/http/http_status_code.h"
+
+namespace brave_rewards::internal::endpoints {
+using Error = GetWallet::Error;
+using Result = GetWallet::Result;
+
+namespace {
+
+Result ParseBody(const std::string& body) {
+  auto value = base::JSONReader::Read(body);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Failed to parse body!");
+    return base::unexpected(Error::kFailedToParseBody);
+  }
+
+  auto* deposit_account_provider =
+      value->GetDict().FindDict("depositAccountProvider");
+  if (!deposit_account_provider) {
+    return {};
```

Comment:
Is this an error case? If not, then maybe explicitly providing the pair arguments would help clarify?